routing/payment_session.go

```diff
@@ -243,6 +243,15 @@ func (p *paymentSession) RequestRoute(maxAmt, feeLimit lnwire.MilliSatoshi,
 
 		switch {
 		case err == errNoPathFound:
+			// Don't split if this is a legacy payment without mpp
+			// record.
+			if p.payment.PaymentAddr == nil {
```

Comment:
any reason not to check the dest features instead?

Reply:
I think it is still possible to set dest features but not have a payment addr. In that case, we still can't split?